src/engine.rs

```diff
@@ -121,17 +121,25 @@ impl Engine {
     }
 
     /// Increases the balance for a given address.
-    pub fn increase_balance(address: &Address, amount: &U256) {
-        let mut balance = Self::get_balance(address);
-        balance += *amount;
-        Self::set_balance(address, &balance);
+    fn increase_balance(address: &Address, amount: &U256) -> Result<(), ExitError> {
+        let balance = Self::get_balance(address);
+        let (new_balance, overflow) = balance.overflowing_add(*amount);
+        if overflow {
+            return Err(ExitError::StackOverflow);
```

Comment:
This is not a stack overflow, it is just arithmetic overflow. It doesn't look like there is an explicit enum variant which supports this error, so I guess the right thing to do would be to use `ExitError::Other`.

Reply:
Ah, sorry was late, STACK overflow, of course.